shared/AppInsightsCore/src/JavaScriptSDK/CoreUtils.ts

```diff
@@ -466,3 +484,64 @@ export class EventHelper {
      */
     public static DetachEvent: (obj: any, eventNameWithoutOn: string, handlerRef: any) => void = _detachEvent;
 }
+
+const doPerfActiveKey = "CoreUtils.doPerf";
+
+/**
+ * Helper function to wrap a function with a perf event
+ * @param mgrSource - The Performance Manager or a Performance provider source (may be null)
+ * @param getSource - The callback to create the source name for the event (if perf monitoring is enabled)
+ * @param func - The function to call and measure
+ * @param details - A function to return the payload details
+ * @param isAsync - Is the event / function being call asynchronously or synchronously 
+ */
+export function doPerf<T>(mgrSource: IPerfManagerProvider | IPerfManager, getSource: () => string, func: (perfEvt?: IPerfEvent) => T, details?: () => any, isAsync?: boolean) {
+    if (mgrSource) {
+        let perfMgr: IPerfManager = mgrSource as IPerfManager;
+        if (perfMgr && CoreUtils.isFunction(perfMgr["getPerfMgr"])) {
+            // Looks like a perf manager provider object
+            perfMgr = perfMgr["getPerfMgr"]()
+        }
+        
+        if (perfMgr) {
+            let perfEvt: IPerfEvent;
+            let currentActive: IPerfEvent = perfMgr.getCtx(doPerfActiveKey);
+            try {
+                let payload = CoreUtils.isFunction(details) ? details() : null;
+                perfEvt = perfMgr.create(getSource(), payload, isAsync);
+                if (perfEvt) {
+                    if (currentActive && perfEvt.setCtx) {
+                        perfEvt.setCtx(PerfEvent.ParentContextKey, currentActive);
+                        if (currentActive.getCtx && currentActive.setCtx) {
+                            let children: IPerfEvent[] = currentActive.getCtx(PerfEvent.ChildrenContextKey);
+                            if (!children) {
+                                children = [];
+                                currentActive.setCtx(PerfEvent.ChildrenContextKey, children);
+                            }
+    
+                            children.push(perfEvt);
+                        }
+                    }
+    
+                    // Set this event as the active event now
+                    perfMgr.setCtx(doPerfActiveKey, perfEvt);
+                    return func(perfEvt);
+                }
+            } catch (ex) {
+                if (perfEvt) {
```

Comment:
Possibly also want to check that setCtx is defined here

Reply:
Fixed